src/ethereum_spec_tools/new_fork.py

```diff
@@ -0,0 +1,169 @@
+"""
+Tool to create a new fork using the latest fork
+"""
+
+import argparse
+import fnmatch
+import os
+
+DESCRIPTION = """
+Creates the base code for a new fork by using the \
+existing code from a given fork.
+
+
+The command takes 4 arguments, 2 of which are optional
+    from_fork: The fork name from which the code is to be duplicated \
+Eg - "Tangerine Whistle"
+    to_fork: The fork name of the new fork Eg - "Spurious Dragon"
+    from_test(Optional): Name of the from fork within the test fixtures \
+in case it is different from fork name Eg - "EIP150"
+    to_test(Optional): Name of the to fork within the test fixtures \
+in case it is different from fork name Eg - "EIP158"
+
+
+If one wants to create the spurious dragon fork fron the tangerine whistle one
+    python src/ethereum_spec_tools/new_fork.py \
+--from_fork="Tangerine Whistle" \
+--to_fork="Spurious Dragon" \
+--from_test=EIP150 \
+--to_test=EIP158
+
+The following will have to however, be updated manually
+    1. The fork number and MAINNET_FORK_BLOCK in __init__.py
+    2. Any absolute package imports from other forks eg. in trie.py
+    3. Package Names under setup.cfg
+"""
+
+parser = argparse.ArgumentParser(
+    description=DESCRIPTION,
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+)
+
+parser.add_argument("--from_fork", dest="from_fork", type=str, required=True)
+parser.add_argument("--to_fork", dest="to_fork", type=str, required=True)
+parser.add_argument("--from_test", dest="from_test", type=str)
+parser.add_argument("--to_test", dest="to_test", type=str)
+
+
+def find_replace(dir: str, find: str, replace: str, file_pattern: str) -> None:
+    """
+    Replace the occurrance of a certain text in files with a new text
+    """
+    for path, _, files in os.walk(dir):
+        for filename in fnmatch.filter(files, file_pattern):
+            file_path = os.path.join(path, filename)
+            with open(file_path) as f:
+                s = f.read()
+            s = s.replace(find, replace)
+            with open(file_path, "w") as f:
+                f.write(s)
```

Comment:
Couple suggestions here:

First, is to use binary mode (`b`), because otherwise `\n` may get replaced with `\r\n` on windows platforms.

Second is to avoid reopening the same file, and to use (read-update) `r+` mode instead. Slightly safer if something else is modifying the file at the same time.

```suggestion
            with open(file_path, "r+b") as f:
                s = f.read()
                s = s.replace(find, replace)
                f.seek(0) # I don't know if this is necessary.
                f.write(s)
```

Reply:
added